tests/rest/client/test_relations.py

```diff
@@ -1037,30 +1040,52 @@ def test_thread(self) -> None:
         """
         Test that threads get correctly bundled.
         """
-        self._send_relation(RelationTypes.THREAD, "m.room.test")
-        channel = self._send_relation(RelationTypes.THREAD, "m.room.test")
+        self._send_relation(
+            RelationTypes.THREAD, "m.room.test", access_token=self.user2_token
+        )
+        channel = self._send_relation(
+            RelationTypes.THREAD, "m.room.test", access_token=self.user2_token
+        )
         thread_2 = channel.json_body["event_id"]
 
-        def assert_thread(bundled_aggregations: JsonDict) -> None:
-            self.assertEqual(2, bundled_aggregations.get("count"))
-            self.assertTrue(bundled_aggregations.get("current_user_participated"))
-            # The latest thread event has some fields that don't matter.
-            self.assert_dict(
-                {
-                    "content": {
-                        "m.relates_to": {
-                            "event_id": self.parent_id,
-                            "rel_type": RelationTypes.THREAD,
-                        }
+        def _gen_assert(participated: bool) -> Callable[[JsonDict], None]:
+            def assert_thread(bundled_aggregations: JsonDict) -> None:
+                self.assertEqual(2, bundled_aggregations.get("count"))
+                self.assertEqual(
+                    participated, bundled_aggregations.get("current_user_participated")
+                )
+                # The latest thread event has some fields that don't matter.
+                self.assert_dict(
+                    {
+                        "content": {
+                            "m.relates_to": {
+                                "event_id": self.parent_id,
+                                "rel_type": RelationTypes.THREAD,
+                            }
+                        },
+                        "event_id": thread_2,
+                        "sender": self.user2_id,
+                        "type": "m.room.test",
                     },
-                    "event_id": thread_2,
-                    "sender": self.user_id,
-                    "type": "m.room.test",
-                },
-                bundled_aggregations.get("latest_event"),
-            )
+                    bundled_aggregations.get("latest_event"),
+                )
+
+            return assert_thread
 
-        self._test_bundled_aggregations(RelationTypes.THREAD, assert_thread, 10)
+        # A user which has sent the root event or replied has participated.
+        self._test_bundled_aggregations(RelationTypes.THREAD, _gen_assert(True), 9)
```

Comment:
The shared bits of these tests make it pretty hard to follow. "user 1" participated because `_test_bundled_aggregations` uses `self.parent_id` (the thread root event) by default which was sent by `self.user_token` in `prepare`.

This test would be a lot more clear if we passed in the root event ID and token explicitly.

We could call the token `user_who_sent_parent_event_token` for the person who sent `parent_event_id` to make an even stronger connection.

Reply:
I attempted to clarify this a bit with comments in b88cac08976af7f49ccddc9a2a2583dbeff4cf8b.